core/src/main/java/org/apache/iceberg/PropertiesUpdate.java

```diff
@@ -89,6 +89,11 @@ public UpdateProperties defaultFormat(FileFormat format) {
 
     newProperties.putAll(updates);
 
+    // Validate the metrics
+    if (base != null && base.schema() != null) {
+      MetricsConfig.fromProperties(newProperties).validateReferencedColumns(base.schema());
```

Comment:
I don't think this is tested. Could you add a test to the property update tests?

Reply:
This is covered in the same test as the other one (the exception is on a different line though):

```
org.apache.iceberg.TestSchemaAndMappingUpdate > testModificationWithMetricsMetrics[formatVersion = 1] FAILED
    java.lang.AssertionError: No exception was thrown (Creating metrics for non-existent column fails), expected: org.apache.iceberg.exceptions.ValidationException
        at org.junit.Assert.fail(Assert.java:88)
        at org.apache.iceberg.AssertHelpers.assertThrows(AssertHelpers.java:65)
        at org.apache.iceberg.TestSchemaAndMappingUpdate.testModificationWithMetricsMetrics(TestSchemaAndMappingUpdate.java:169)

org.apache.iceberg.TestSchemaAndMappingUpdate > testModificationWithMetricsMetrics[formatVersion = 2] FAILED
    java.lang.AssertionError: No exception was thrown (Creating metrics for non-existent column fails), expected: org.apache.iceberg.exceptions.ValidationException
        at org.junit.Assert.fail(Assert.java:88)
        at org.apache.iceberg.AssertHelpers.assertThrows(AssertHelpers.java:65)
        at org.apache.iceberg.TestSchemaAndMappingUpdate.testModificationWithMetricsMetrics(TestSchemaAndMappingUpdate.java:169)

```